app/services/jwt_token.rb

```diff
@@ -1,7 +1,7 @@
 require 'jwt'
 require 'base64'
 
-class JwtToken < Struct.new(:token)
+class JwtToken < Struct.new(:token, :url, :client_id)
```

Comment:
I think we should create a subclass of `JwtToken` for the Keycloak use case.

Reply:
@timogoebel Can you please eloborate why we should create a subclass of the JwtToken.

I have kept it in the same class becasue the `:url` and  `:client_id` seem to be generic to all JWT's. We us them for them for [validation of the token](https://auth0.com/docs/api-auth/tutorials/verify-access-token#verify-the-signature).

In hammer we set the two params:
1. [`url`](https://github.com/theforeman/hammer-cli-foreman/pull/405/files#diff-ee28c580d4029a2e5b5bd753289594a2R20) from where we can get the public key that we use to verify signature as we are going [here](https://github.com/theforeman/foreman/pull/6549/files#diff-40a09b2d31a74a6001b62290a4683eb4R30).
2. [`client_id`](https://github.com/theforeman/hammer-cli-foreman/pull/405/files#diff-ee28c580d4029a2e5b5bd753289594a2R21) that we use to verify token[aud] as we have done [here](https://github.com/theforeman/foreman/pull/6549/files#diff-40a09b2d31a74a6001b62290a4683eb4R51).

Reply:
The JWT we'll issue from Foreman does not have this payload.

The graphql function to sign in a user will look something like this:

```ruby
# frozen_string_literal: true

class Mutations::SignInUser < GraphQL::Function
  argument :login, InputTypes::AuthProviderLoginInputType

  type do
    name 'SIGN_IN_USER_PAYLOAD'

    field :user_id, types.ID
    field :token, types.String
    field :fullname, types.String
  end

  def call(_obj, args, _ctx)
    login = args[:login] || {}
    username = login[:username]
    password = login[:password]
    return unless username && password

    user = User.try_to_login(username, password)
    return unless user

    OpenStruct.new(user_id: user.id, token: user.jwt_token!, fullname: user.fullname)
  end
end
```

I hope that makes clear how my JWT is supposed to be used.

Reply:
@timogoebel  thanks for the explanation :)